rosetta/transactions/intent.go

```diff
@@ -0,0 +1,155 @@
+// Copyright 2021 Optakt Labs OÜ
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not
+// use this file except in compliance with the License. You may obtain a copy of
+// the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+// WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+// License for the specific language governing permissions and limitations under
+// the License.
+
+package transactions
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/onflow/cadence"
+	"github.com/onflow/flow-go/model/flow"
+
+	"github.com/optakt/flow-dps/models/convert"
+	"github.com/optakt/flow-dps/models/dps"
+	"github.com/optakt/flow-dps/rosetta/failure"
+	"github.com/optakt/flow-dps/rosetta/object"
+)
+
+// Intent describes the intent of an array of Rosetta operations.
+type Intent struct {
+	From     flow.Address
+	To       flow.Address
+	Amount   cadence.UFix64
+	Payer    flow.Address
+	Proposer flow.Address
+
+	ReferenceBlock            flow.Identifier
+	ProposerKeySequenceNumber uint64
+	GasLimit                  uint64
+}
+
+// CreateTransactionIntent creates a transaction Intent from two operations given as input.
+// Specified operations should be symmetrical, a deposit and a withdrawal from two
+// different accounts. At the moment, the only fields taken into account are the
+// account IDs, amounts and type of operation.
+func (p *Parser) CreateTransactionIntent(operations []object.Operation) (*Intent, error) {
+
+	send := operations[0]
+	receive := operations[1]
+
+	// The amount is the same, but for the 'send' operation the amount will have the '-' prefix.
+	// If it doesn't it means that the operations should be switched.
+	if !strings.HasPrefix(send.Amount.Value, "-") {
+		receive = operations[0]
+		send = operations[1]
+
+		// Now that we have switched operations, make sure that the other operation
+		// is in fact negative.
+		if !strings.HasPrefix(send.Amount.Value, "-") {
+			return nil, fmt.Errorf("invalid amounts - deposit operation missing")
+		}
+	}
```

Comment:
```suggestion
    switch {
    	case strings.HasPrefix(operations[0].Amount.Value, "-"):
			  receive = operations[0]
			  send = operations[1]
		
    	case strings.HasPrefix(operations[1].Amount.Value, "-"):
			  receive = operations[1]
			  send = operations[0]
    	
    	default:
			return nil, fmt.Errorf("invalid amounts - deposit operation missing")
    }
```

WDYT of something like this? I feel like it's clearer

Reply:
Also what do we do in the case where both are negative?